src/Microsoft.Data.Analysis/DataFrame.Join.cs

```diff
@@ -423,5 +448,25 @@ public DataFrame Merge(DataFrame other, string[] leftJoinColumns, string[] right
 
             return ret;
         }
+
+        private static bool IsLeftDataFrameRetained(DataFrame retainedDataFrame, DataFrame supplementaryDataFrame, JoinAlgorithm joinAlgorithm)
```

Comment:
nit: this can be 

```C#
private static bool IsLeftDataFrameRetained(DataFrame retainedDataFrame, DataFrame supplementaryDataFrame, JoinAlgorithm joinAlgorithm) =>
            joinAlgorithm switch
            {
                JoinAlgorithm.Left or JoinAlgorithm.FullOuter => true,
                JoinAlgorithm.Right => false,
                JoinAlgorithm.Inner => retainedDataFrame.Rows.Count > supplementaryDataFrame.Rows.Count,
                _ => throw new InvalidOperationException(nameof(joinAlgorithm)),
            };
```